sdk/python/kfp/dsl/_component.py

```diff
@@ -143,13 +143,13 @@ def flip_component(flip_result):
 
   @wraps(func)
   def _graph_component(*args, **kargs):
-    # We need to make sure that the arguments are correctly mapped to inputs
-    # regardless of the passing order
-    signature = inspect.signature(func)
-    bound_arguments = signature.bind(*args, **kargs)
     graph_ops_group = Graph(func.__name__)
-    graph_ops_group.inputs = list(bound_arguments.arguments.values())
-    graph_ops_group.arguments = bound_arguments.arguments
+    graph_ops_group.inputs = list(args) + list(kargs.values())
```

Comment:
I'm not sure this is correct. The ordering of `kargs.values()` is not guaranteed to be stable and more importantly it's not guaranteed to correspond to the input order. This is the reason why we're using `signature.bind`.